packages/react-native-editor/bin/test-e2e.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash -e
+
+set -o pipefail
+
+# If TEST_RN_PLATFORM is empty or equal to "android", launch the Android emulator.
+if [ -z "$TEST_RN_PLATFORM" ] || [ "$TEST_RN_PLATFORM" = "android" ]; then
```

Comment:
One thing I noticed is that this is being executed in [GitHub actions](https://github.com/WordPress/gutenberg/actions/runs/6668614152/job/18124542373?pr=55662#step:10:358) I don't think it's a blocker but we might consider not running it if it's not needed.

Reply:
Noted. I'll explore avoiding this in a follow up PR.

Reply:
Actually, as I consider this more, I am confused. My perception is that these `*:local` scripts were [intended](https://github.com/wordpress-mobile/gutenberg-mobile/commit/dda37b7686b946b2ce35f23ca40c145fcc1bf220) to be used for running the tests on a local development machine, not the CI. 

It is odd to me that the GitHub CI workflows currently leverage this script. 

https://github.com/WordPress/gutenberg/blob/c4903d61825fa9f04c3f1ead3866378d80c4a1a2/.github/workflows/rnmobile-android-runner.yml#L81-L82

https://github.com/WordPress/gutenberg/blob/c4903d61825fa9f04c3f1ead3866378d80c4a1a2/.github/workflows/rnmobile-ios-runner.yml#L79-L80

There are [GitHub](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) and [Buildkite](https://buildkite.com/docs/pipelines/environment-variables#buildkite-environment-variables) environment variables we could use to conditionally run the emulator portion of the scripts, but the latter is not relevant to this repository. It feels more appropriate to avoid running the `*:local` scripts altogether or update the abstracted script to not be local-specific. 

WDYT?

Reply:
I agree it is confusing, the understanding I have of the `local` scripts is to be able to build and run the tests in the same machine, regardless if it's CI or a development machine. I guess that we needed something different for SauceLabs since we would make a build of the demo app, upload it, and then run it using their service, where they have things like the WDA driver already installed in their simulators.

As for GitHub actions, we need to build and run the tests on the same machine as we do in our development machines.

I think once we migrate to Buildkite fully, we could have just one script to build/run E2E tests regardless of the environment. I think there's quite a bit of duplication between Gutenberg Mobile and Gutenberg's scripts so working on reducing duplication and confusion it'd be a great goal.

Reply:
Alright. Thank you for the context and guidance. I addressed this in https://github.com/WordPress/gutenberg/pull/55708.